src/dmd/dmodule.d

```diff
@@ -634,9 +634,11 @@ extern (C++) final class Module : Package
         {
             if (!strcmp(srcfile.toChars(), "object.d"))
             {
-                .error(loc, "cannot find source code for runtime library file 'object.d'");
-                errorSupplemental(loc, "dmd might not be correctly installed. Run 'dmd -man' for installation instructions.");
-                errorSupplemental(loc, "config file: %s", FileName.canonicalName(global.inifilename));
+                // object.d is automatically imported.  If object.d doesn't exist, it's not necessarily an error.
+                // For example, the user could be creating a library in D that is intened to be called from
+                // another language and, therefore, does not wish to use features of the D runtime.  We
+                // return `false` here and rely on future semantic processing to catch any errors.
+                return false;
```

Comment:
I suspect this will make bad installations less obvious, especially to newcomers, that don't know anything about what's to be expected in object.d.

Also, in combination with optional TypeInfo, ModuleInfo and Throwable, this might silently and advertently build unexpected code due to missing import paths.

If someone is actually starting the adventure of porting to another platform, adding a switch or even creating an empty object.d should not be that much of an issue.

Reply:
> I suspect this will make bad installations less obvious, especially to newcomers, that don't know anything about what's to be expected in object.d.

I doubt it.  With basic things like `Object`, `size_t`, and `string` defined in `object.d`, they'll know right away that something's not right.  The existing error message is quite useless to a newcomer anyway. Because `object.d` is imported behind the scenes, they won't even know what the compiler's talking about, let alone what to do about it.  This PR is no worse than the status quo in the case of a bad installation, IMO.

> Also, in combination with optional TypeInfo, ModuleInfo and Throwable, this might silently and advertently build unexpected code due to missing import paths.

I'm not sure what you mean here.  The compiler will fail if it can't find needed symbols, and the linker will fail if it can't find needed references.  The compiler should succeed if it found everything it needed.  The likelihood of an build that shouldn't have succeeded as a result of this PR seems extremely rare to me.

> If someone is actually starting the adventure of porting to another platform, adding a switch or even creating an empty object.d should not be that much of an issue.

Agreed, but this PR is primarily for those who are writing libraries in D to be consumed by other languages.  There is no benefit that I can see to forcing them to add an empty `object.d` to their source code.  It's just unnecessary friction and cruft.

Reply:
> This PR is no worse than the status quo in the case of a bad installation, IMO.

I disagree: without the current error, you are likely to get undefined identifiers/unknown properties for symbols mostly considered part of the language and no hint what actually is wrong.



> The compiler will fail if it can't find needed symbols,

For example, this silently produces different results when compiled separately with or without object.d:
```
module noobj;
extern(C) int printf(const char*, ...);
extern(C) void goo();

void foo()
{
	scope(exit) printf("exit\n");
	printf("foo\n");
	goo();
}
```
when linked afterwards with this compiled with object.d:
```
import noobj;

extern(C) void goo()
{
	throw new Exception("ho");
}

void main()
{
	foo();
}
```

> There is no benefit that I can see to forcing them to add an empty object.d to their source code.

Agreed, but a command line switch might do the trick here. Maybe even couple with -betterC?

Reply:
> Because object.d is imported behind the scenes, they won't even know what the compiler's talking about

A slightly modified error message including "implicitly imported" might help in this case.

BTW: object.d can also be imported explicitly, and failing to find it should still raise an error.